web/src/billing/event_status.ts

```diff
@@ -82,41 +105,70 @@ export function initialize_retry_with_another_card_link_click_handler() {
             [],
             "POST",
             (response) => {
-                window.location.replace(response.stripe_session_url);
+                const response_schema = z.object({
+                    stripe_session_url: z.string(),
+                });
```

Comment:
This particular schema can be moved into `helpers.ts`, because it is shared between `event_status.ts` and `upgrade.ts`.